src/traces/heatmap/convert_column_xyz.js

```diff
@@ -11,47 +11,66 @@
 
 var Lib = require('../../lib');
 
+module.exports = function convertColumnData(trace, ax1, ax2, var1Name, var2Name, arrayVarNames) {
```

Comment:
I think that should be straightforward. I've already implemented calcPriority to reorder the calc step. Calling this is one of the last steps in carpet defaults, so it should be possible to simply move this to the beginning of `calc` instead. It might cascade into `scattercarpet` and `contourcarpet`, but I think the only downside is the inability to bail out early due to invalid data.